### PR TITLE
fpgad: add config file processing

### PR DESCRIPTION
**Pull request by tswhison**
Parse the given (--config,-c) configuration file, constructing a
supported devices table based on its contents. If parsing fails,
revert to a default supported devices table.

### CODE REVIEW

File: tools/base/fpgad/config_file.c [hunk 1]
```diff
@@ -179,7 +183,299 @@ STATIC cfg_plugin_configuration *alloc_configuration(bool enabled,
...
+
+		if (!head) {
+			head = alloc_device(vendor_id, device_id);
+			if (!head) {
+				LOG("malloc failed.\n");
+				goto out_free;
+			}
+
+			id = head;
+		} else {
```
Comment (jelonanX) [inline tools/base/fpgad/config_file.c]: Head is initialized to null, would it ever get into the else?

Comment (tswhison) [reply tools/base/fpgad/config_file.c]: This is within the for loop, so !head will only be true the first time through.  Subsequent iterations will it the else case and append to the end of the list via id.

Comment (jelonanX) [reply tools/base/fpgad/config_file.c]: 👍

File: tools/base/fpgad/config_file.c [hunk 2]
```diff
@@ -183,7 +187,299 @@ STATIC cfg_plugin_configuration *alloc_configuration(bool enabled,
...
+			goto out_free;
+		}
+
+		if (json_object_array_length(j_dev) != 2) {
+			LOG("%s 'devices' entry %d not array[2].\n",
+				name, i);
+			goto out_free;
+		}
+
+		j_vid = json_object_array_get_idx(j_dev, 0);
```
Comment (r-rojo) [inline tools/base/fpgad/config_file.c]: You think this (getting or parsing an int from a jarray at index I) could be refactored into a function?

Comment (tswhison) [reply tools/base/fpgad/config_file.c]: Sure.. but for 2 items, I'm not sure it's worth the trouble.

**Pull request merged**